process/block/baseProcess.go

```diff
@@ -2352,10 +2352,19 @@ func (bp *baseProcessor) requestProofIfNeeded(currentHeaderHash []byte, epoch ui
 		return false
 	}
 	if bp.proofsPool.HasProof(shardID, currentHeaderHash) {
-		bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)].hasProof = true
+		_, ok := bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)]
+		if ok {
+			bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)].hasProof = true
+		}
+
 		return true
 	}
 
+	_, ok := bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)]
+	if !ok {
+		bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)] = &hdrInfo{}
```

Comment:
this can introduce panics on accessing the hdrInfo.hdr, as it is always required to have it set

Reply:
pushed